.github/workflows/e2e-main.yml

```diff
@@ -0,0 +1,142 @@
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: e2e-tests-main
+
+on:
+ schedule:
+    - cron: '0 0 * * *'
+ workflow_dispatch:
+    inputs:
+      organization:
+        default: 'crc-org'
+        description: 'CRC Runs Containers'
+        type: string
+        required: true
+      repositoryName:
+        default: 'crc-extension'
+        description: 'Podman Desktop OpenShift Local Extension repository name'
+        type: string
+        required: true
+      branch:
+        default: 'main'
+        description: 'Podman Desktop OpenShift Local Extension repo branch'
+        type: string
+        required: true
+      mode:
+        description: 'Build mode for Podman Desktop'
+        type: choice
+        options:
+          - production
+          - development
+          
+jobs:
+  e2e-tests:
+    name: Podman Desktop OpenShift Local Extension E2E tests
+    runs-on: ubuntu-24.04
+    env: 
+      MODE: ${{ inputs.mode || 'production' }}
+      REPO: ${{ inputs.repositoryName || 'crc-extension' }}
+      ORGANIZATION: ${{ inputs.organization || 'crc-org' }}
+      BRANCH: ${{ inputs.branch || 'main' }}
+    steps:
+      - uses: actions/checkout@v4
+        if: github.event_name == 'workflow_dispatch' || github.event_name == 'workflow_run'
+        with:
+          repository: ${{ env.ORGANIZATION }}/${{ env.REPO }}
+          ref: ${{ env.BRANCH }}
+          path: ${{ env.REPO }}
+
+      - uses: actions/checkout@v4
+        if: github.event_name == 'push'
+        with: 
+          path: crc-extension
+
+      - name: Setup default repository name
+        env: 
+          EVENT: ${{ github.event_name }}
+        run: |
+          repository=crc-extension
+          if [[ "$EVENT" == 'workflow_dispatch' || "$EVENT" == 'workflow_run' ]]; then
+            repository=${{ env.REPO }}
+          fi
+          echo "REPOSITORY=$repository" >> $GITHUB_ENV
+          
+      # Checkout podman desktop
+      - uses: actions/checkout@v4
+        with:
+          repository: containers/podman-desktop
+          ref: main
+          path: podman-desktop
+
+      - uses: pnpm/action-setup@v4
+        name: Install pnpm
+        with:
+          run_install: false
+          package_json_file: ./${{ env.REPO }}/package.json
```

Comment:
instead of env.REPO use `podman-desktop` as we need to install pnpm in podman-desktop to build it.